File: VB/Grid_DateOccurring/DataGenerator.vb
Imports System
Imports System.Collections.Generic
Imports System.Threading

Namespace Grid_DateOccurring
    Friend Class DataGenerator
        Public Property [Date]() As Date
        Public Property Sales() As Decimal

        Public Shared Function CreateData() As List(Of DataGenerator)
            Dim data As New List(Of DataGenerator)()

            For i As Integer = 0 To 99
                Dim record As New DataGenerator()
                Dim seed As Integer = CLng(Date.Now.Ticks) And &HFFFF
                Dim startDate As New Date(Date.Today.Year, 1, 1)
                Dim rand As New Random()
                Dim range As Integer = (Date.Today.Subtract(startDate)).Days

                record.Sales = (New Random(seed)).Next(0, 1000)
                record.Date = startDate.AddDays(rand.Next(range))
                data.Add(record)
                Thread.Sleep(3)
            Next i
            Return data
        End Function
    End Class
End Namespace

File: CS/Grid_DateOccurring/DataGenerator.cs
using System;
using System.Collections.Generic;
using System.Threading;

namespace Grid_DateOccurring {
    class DataGenerator {
        public DateTime Date { get; set; }
        public decimal Sales { get; set; }        

        public static List<DataGenerator> CreateData() {
            List<DataGenerator> data = new List<DataGenerator>();

            for (int i = 0; i < 100; i++) {
                DataGenerator record = new DataGenerator();
                int seed = (int)DateTime.Now.Ticks & 0x0000FFFF;
                DateTime startDate = new DateTime(DateTime.Today.Year, 1, 1);
                Random rand = new Random();
                int range = (DateTime.Today - startDate).Days;  

                record.Sales = new Random(seed).Next(0, 1000);
                record.Date = startDate.AddDays(rand.Next(range));
                data.Add(record);
                Thread.Sleep(3);
            }
            return data;
        }
    }
}
